### PR TITLE
Fix Mailjet API Call

### PR DESCRIPTION
**Pull request by ottaviano**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -326,11 +326,11 @@ en_marche_mailer:
         # can use same old sound connexions settings and will be prepend
     mail_aggregator:
         routing_keys:
-            - 'em_mails_*' # default
+            - 'em_mails.*.*' # default
             # or
-            #- 'em_mails_campaign_*
-            #- 'em_mails_transactional_*
-            #- 'em_mails_transactional_app_name
+            #- 'em_mails.campaign.*
```
Comment (Remg) [inline README.md]: missing end quotes

File: README.md [hunk 2]
```diff
@@ -326,11 +326,11 @@ en_marche_mailer:
...
+            #- 'em_mails.transactional.*
+            #- 'em_mails.transactional.app_name
```
Comment (Remg) [inline README.md]: should it be `em_mails.transactional.app_name.*` ?

File: src/DependencyInjection/Configuration.php [hunk 3]
```diff
@@ -172,7 +172,7 @@ private function addMailAggregatorSection(ArrayNodeDefinition $rootNode): void
                     ->children()
                         ->arrayNode('routing_keys')
                             ->info('Defines the rules that let the MailConsumer listen to routes.')
-                            ->defaultValue(['em_mails_*'])
+                            ->defaultValue(['em_mails.*.*'])
```
Comment (thomasbisignani) [inline src/DependencyInjection/Configuration.php]: You can update the `scalarPrototype()->info()` message : 
```
->info('Rule to define routes to listen to. i.e "em_mails.*.*" ...)
```

File: src/DependencyInjection/Configuration.php [hunk 4]
```diff
@@ -200,7 +200,7 @@ private function addMailApiProxySection(ArrayNodeDefinition $rootNode): void
                     ->children()
                         ->arrayNode('routing_keys')
                             ->info('Defines the rules that let the MailRequestConsumer listen to routes.')
-                            ->defaultValue(['em_mail_requests_*'])
+                            ->defaultValue(['em_mail_requests.*.*'])
```
Comment (thomasbisignani) [inline src/DependencyInjection/Configuration.php]: Same, you can update this `scalarPrototype()->info()` message.

IMO the message info is not correct, what do you think to add the `em_mail_requests` prefix to match `em_mail_requests.{type}.{app_name}` ? 

Can you fix it in this PR ?

File: README.md (hunk 2 above)
Comment (ottaviano) [reply README.md]: nop, just 3 levels, prefix, mail type and app_name
https://github.com/EnMarche/mailer-bundle/blob/6025591f94b42f6dd704a33eb1ee868f7832d476/src/Mailer/Transporter/AmqpMailTransporter.php#L53-L60

File: composer.json [hunk 5]
```diff
@@ -3,18 +3,21 @@
     "license": "GPL-3.0",
     "description": "Mailer bundle to ease proxying SAAS among private applications.",
     "type": "symfony-bundle",
+    "config": {
+        "sort-packages": true
+    },
     "require": {
         "php": ">=7.1",
+        "guzzlehttp/guzzle": "^6.3",
```
[the remark this answers is not in the document]
Comment (ottaviano) [reply composer.json]: moved on `dev` 👍

File: composer.json [hunk 6]
```diff
@@ -3,18 +3,21 @@
...
+        "php-amqplib/rabbitmq-bundle": "^1.14",
         "ramsey/uuid": "^3.0",
+        "ramsey/uuid-doctrine": "^1.5",
```
[the remark this answers is not in the document]
Comment (ottaviano) [reply composer.json]: done

File: composer.json [hunk 7]
```diff
@@ -3,18 +3,21 @@
     "license": "GPL-3.0",
     "description": "Mailer bundle to ease proxying SAAS among private applications.",
     "type": "symfony-bundle",
+    "config": {
+        "sort-packages": true
+    },
     "require": {
         "php": ">=7.1",
+        "guzzlehttp/guzzle": "^6.3",
+        "php-amqplib/rabbitmq-bundle": "^1.14",
```
[the remark this answers is not in the document]
Comment (ottaviano) [reply composer.json]: @HeahDude, moved in `dev` part

**Comment by thomasbisignani** [comment]
:+1:

File: src/DependencyInjection/Configuration.php (hunk 3 above)
Comment (ottaviano) [reply src/DependencyInjection/Configuration.php]: fixed, thanks 👍

File: src/DependencyInjection/Configuration.php (hunk 4 above)
Comment (ottaviano) [reply src/DependencyInjection/Configuration.php]: fixed too

File: src/Repository/AddressRepository.php [hunk 8]
```diff
@@ -22,10 +23,10 @@ public function findOneForEmail(string $email): ?Address
     }
 
     /**
-     * @throws \Doctrine\ORM\NonUniqueResultException
+     * @throws NonUniqueResultException
      */
     public function findOneForRecipient(RecipientInterface $recipient)
     {
-        return $this->findOneForEmail($recipient->getEmail(), $recipient->getName());
+        return $this->findOneForEmail($recipient->getEmail());
```
[the remark this answers is not in the document]
Comment (HeahDude) [reply src/Repository/AddressRepository.php]: Hmmm, in fact, https://github.com/EnMarche/mailer-bundle/blob/master/src/Repository/AddressRepository.php#L27 should be used instead.

[the remark this answers is not in the document]
Comment (HeahDude) [reply src/Repository/AddressRepository.php]: Ah no sorry, I though this was called from the factory. Nevermind.

[the remark this answers is not in the document]
Comment (HeahDude) [reply src/Repository/AddressRepository.php]: Yet this needs a fix, the name should be used in the query.

[the remark this answers is not in the document]
Comment (ottaviano) [reply src/Repository/AddressRepository.php]: `findOneForEmail` handles only one argument, but I have already fixed it in the next PR (by adding `name` argument)

**Comment by ottaviano** [comment]
[ch5535]

**Pull request merged**